components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml

```diff
@@ -94,12 +94,35 @@
   #set ($collaborators = $request.getParameterValues('collaborator'))
   #set ($accessLevels = $request.getParameterValues('accessLevel'))
   #set ($owner = "$!{request.get('owner')}")
+  #if ("$!{owner}" != "")
+    ##check the validity of the new owner
+    #set ($ownerDocRef = $services.model.resolveDocument("$owner"))
+    #if ($xwiki.getDocument($ownerDocRef).getObject("PhenoTips.PhenoTipsGroupClass"))
+      #set ($validGroup = true)
+      ## Check to see if user has permission to transfer ownership
+      #set ($adminGroup = $owner + " Administrators")
+      #if (!$xwiki.getUser().isUserInGroup($owner) &amp;&amp; !$xwiki.getUser().isUserInGroup($adminGroup) &amp;&amp; !$hasAdmin)
+        #set ($cannotTransfer = true)
+      #end
+    #elseif ($services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.name = :ownerName").bindValue("ownerName", "$owner").execute().count() &gt; 0)
+      #set ($validUser = true)
+    #end
+    #if (!($validUser || $validGroup))
+      #set ($invalidOwner = true)
+    #end
+  #end
   #if (!$patient)
     #set ($discard = $response.setStatus(400))## no patient : bad request
     Could not find patient with identifier "$!{request.patient}", cannot complete the update action.
-  #elseif($collaborators.size() != $accessLevels.size())
+  #elseif ($collaborators.size() != $accessLevels.size())
     #set ($discard = $response.setStatus(400))## mismatched user/level parameters, bad request
     The request is malformed: could not match each collaborator with exactly one level of access.
+  #elseif ($invalidOwner)
```

Comment:
`$invalidOwner` might not have been initialized. Same for `$cannotTransfer`. Make sure you always initialize variables.

This code here can't have any bad security consequences, but you could actually be hit by a different bug, possibly affecting security, if you check the value of an uninitialized variable that is actually initialized outside your code and is left in the context. You're only setting `invalidOwner` to `true` once here, but it could be set to `true` by a different template, with a different meaning. Since all Velocity variables are global by default, you should always be careful about their initialization.